builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,270 @@
+			},
+			"source_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDbEventSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	name := d.Get("name").(string)
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	sourceIdsSet := d.Get("source_ids").(*schema.Set)
+	sourceIds := make([]*string, sourceIdsSet.Len())
+	for i, sourceId := range sourceIdsSet.List() {
+		sourceIds[i] = aws.String(sourceId.(string))
+	}
+
+	eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+	eventCategories := make([]*string, eventCategoriesSet.Len())
+	for i, eventCategory := range eventCategoriesSet.List() {
+		eventCategories[i] = aws.String(eventCategory.(string))
+	}
+
+	request := &rds.CreateEventSubscriptionInput{
+		SubscriptionName: aws.String(name),
+		SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SourceIds:        sourceIds,
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		EventCategories:  eventCategories,
+		Tags:             tags,
+	}
+
+	log.Println("[DEBUG] Create RDS Event Subscription:", request)
+
+	_, err := rdsconn.CreateEventSubscription(request)
+	if err != nil {
+		return fmt.Errorf("Error creating RDS Event Subscription %s: %s", name, err)
+	}
+
+	log.Println(
+		"[INFO] Waiting for RDS Event Subscription to be ready")
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsDbEventSubscriptionRead(d, meta)
+}
+
+func resourceAwsDbEventSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	sub, err := resourceAwsDbEventSubscriptionRetrieve(d, meta)
+	if err != nil {
+		return err
+	}
+	if sub == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(*sub.CustSubscriptionId)
+	if err := d.Set("name", sub.CustSubscriptionId); err != nil {
+		return err
+	}
+	if err := d.Set("sns_topic", sub.SnsTopicArn); err != nil {
+		return err
+	}
+	if err := d.Set("source_type", sub.SourceType); err != nil {
+		return err
+	}
+	if err := d.Set("enabled", sub.Enabled); err != nil {
+		return err
+	}
+	if err := d.Set("source_ids", flattenStringList(sub.SourceIdsList)); err != nil {
+		return err
+	}
+	if err := d.Set("event_categories", flattenStringList(sub.EventCategoriesList)); err != nil {
+		return err
+	}
+
+	// Set tags, but theyre not in the response
```

Comment:
Tags can be queried via separate API, see how we do it for DB instances:
https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go#L650-L675
The same API seems to be used for many resources, see http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Tagging.html#USER_Tagging.ARN

Also keep in mind we always try to avoid extra API requests for account ID like you can see in https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go#L975 where it wasn't easily unavoidable. **It is avoidable** for RDS Event Subscriptions though since you have `sub.CustomerAwsId` available. 😺